packages/runtime/container-runtime/src/gcSweepReadyUsageDetection.ts

```diff
@@ -0,0 +1,130 @@
+ *  - Throttling the frequency of these crashes via a "Blackout Period" per container per device
+ */
+export class SweepReadyUsageDetectionHandler {
+    private readonly localStorage: Pick<Storage, "getItem" | "setItem">;
+
+    constructor(
+        /** Unique key for this container for diagnostic purposes */
+        private readonly uniqueContainerKey: string,
+        private readonly mc: MonitoringContext,
+        private readonly closeFn: (error?: ICriticalContainerError) => void,
+        localStorageImpl?: Pick<Storage, "getItem" | "setItem">,
+    ) {
+        const noopStorage = { getItem: () => null, setItem: () => {} };
+        if (localStorageImpl !== undefined) {
+            this.localStorage = localStorageImpl;
+        } else {
+            try {
+                // localStorage is not defined in Node environment so this throws
+                this.localStorage = localStorage ?? noopStorage;
+            } catch (error) {
+                this.localStorage = noopStorage;
+            }
+        }
+
+        if (this.localStorage === noopStorage) {
+            // This means the Blackout Period logic will not work.
+            this.mc.logger.sendTelemetryEvent({ eventName: "SweepReadyUsageDetectionHandlerNoOpStorage" });
+        }
+    }
+
+    /**
+      * If SweepReady Usage Detection is enabled, close the main container.
+      * If the "Blackout Period" is set, don't close the container more than once in that period.
+      *
+      * Once Sweep is fully implemented, this will be removed since the objects will be gone
+      * and errors will arise elsewhere in the runtime
+     */
+    public usageDetectedInMainContainer(errorProps: ITelemetryProperties) {
+        if (!sweepReadyUsageDetectionSetting.read(this.mc.config).mainContainer) {
+            return;
+        }
+
+        const pastClosuresMap: Record<string, { lastCloseTime: number; } | undefined> = (() => {
+            try {
+                const rawValue = this.localStorage.getItem(closuresStorageKey);
+                return rawValue === null
+                    ? {}
+                    : JSON.parse(rawValue) as Record<string, { lastCloseTime: number; } | undefined>;
+            } catch (e) {
+                return {};
+            }
+        })();
+
+        const lastCloseTime = pastClosuresMap[this.uniqueContainerKey]?.lastCloseTime;
+        const blackoutPeriodDays = this.mc.config.getNumber(blackoutPeriodDaysKey);
+
+        const error = new SweepReadyUsageError(
+            "SweepReady object used in Non-Summarizer Client",
+            { errorDetails: JSON.stringify({ ...errorProps, lastCloseTime, blackoutPeriodDays }) },
+        );
+
+        // Should close if...
+        const shouldClose = lastCloseTime === undefined // ...We've not closed before, or...
+            || blackoutPeriodDays === undefined // ...there's no blackout duration set, or...
```

Comment:
What's blackoutPeriodDays?

Reply:
A little more details on how this will be used will be helpful :-)

Reply:
Take a look at latest iteration.  Also this function's doc comments mentions the term.  Where would be the best place to explain how this works?

Reply:
If I understand correctly, blackoutPeriodDays is the number of days this error gets ignored if it's not the first time it occurs.

Reply:
Yep.  Will be clearer after this little refactor I think.